include/irr/asset/IImageAssetHandlerBase.h

```diff
@@ -102,16 +102,19 @@ class IImageAssetHandlerBase : public virtual core::IReferenceCounted
 		}
 
 		/*
-			Create an image containing a single row from taking a raw data 
-			to a single row and convert it to any format. Since it's
+			Create an image containing a single row from taking an ICPUBuffer 
+			as a single row and convert it to any format. Since it's
 			data may not only limit to stuff being displayed on a screen,
 			there is an optiomal parameter for bufferRowLength pitch that
 			is helpful while dealing with specific data which needs it.
 		*/
 
 		template<E_FORMAT inputFormat, E_FORMAT outputFormat>
-		static inline core::smart_refctd_ptr<ICPUImage> createSingleRowImageFromRawData(void* rowData, uint32_t texelOrBlockLength, bool createWithBufferRowLengthPitch = false)
+		static inline core::smart_refctd_ptr<ICPUImage> createSingleRowImageFromRawData(core::smart_refctd_ptr<asset::ICPUBuffer> inputBuffer, bool createWithBufferRowLengthPitch = false)
 		{
+			auto rowData = inputBuffer->getPointer();
+			const uint32_t texelOrBlockLength = inputBuffer->getSize() / asset::getTexelOrBlockBytesize(inputFormat);
```

Comment:
ugh, the whole point of taking a `core::smart_refctd_ptr` was to avoid the memcpy, just take over the ownership of the buffer and use it!

Reply:
okay, I will use custom buffer allocator craetion

Reply:
outdated